gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -247,6 +249,88 @@ func (vm *VMKeeper) getGnoStore(ctx sdk.Context) gno.Store {
 	}
 }
 
+// Namespace can be either a user or crypto address.
+var reNamespace = regexp.MustCompile(`^gno.land/(?:r|p)/([\.~_a-zA-Z0-9]+)`)
+
+// checkNamespacePermission check if the user as given has correct permssion to on the given pkg path
+func (vm *VMKeeper) checkNamespacePermission(ctx sdk.Context, creator crypto.Address, pkgPath string) error {
```

Comment:
We should consider removing the admin from the system contract entirely. No one should be able to add a contract under /r/system unless it is voted on by the gov DAO or through a chain upgrade that involves consent from the validators.